### PR TITLE
Refactors and fixes

### PR DESCRIPTION
**Pull request by therealcoder1337**
## PR Type

> What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## What's new?

> List the changes

- Fixed bugs, migrated QDocs to Svelte 5

## Screenshots

> If needed, you can add screenshots here

N/A

## This pull request closes an issue

> Add `Closes`, `Fixes` or `Resolves` followed by `#xxx[,#xxx]` where "xxx" is the issue number

N/A

### CODE REVIEW

File: src/lib/components/private/QDocs.svelte [hunk 1]
```diff
@@ -3,10 +3,23 @@
...
+    componentDocs,
+  }: {
+    children?: Snippet;
+    display?: Snippet;
+    pre?: Snippet;
+    usage?: Snippet;
+    componentDocs: QComponentDocs | QComponentDocs[];
+  } = $props();
+
+  const isDark = $derived($Quaff.dark.isActive);
```
Comment (KilDesu) [inline src/lib/components/private/QDocs.svelte:22]: Use the Quaff class (import from "$lib/classes/Quaff.svelte"): `$derived(Quaff.darkMode.isActive)`

File: src/lib/components/private/QDocs.svelte [hunk 2]
```diff
@@ -46,19 +59,19 @@
...
 
-    {#if $$slots.usage}
+    {#if usage}
       <div class="q-pa-md">
         <div class="heading-usage">
           <h4 class="q-my-xl">Usage</h4>
         </div>
 
-        <slot name="usage" />
+        {@render usage()}
```
Comment (KilDesu) [inline src/lib/components/private/QDocs.svelte:70]: Why not `usage?.()`?

**Review by KilDesu — CHANGES_REQUESTED** [review]
Good, thanks!

File: src/lib/components/private/QDocs.svelte (hunk 2 above)
Comment (therealcoder1337) [reply src/lib/components/private/QDocs.svelte:70]: Because it's surrounded by an `if`.

**Pull request merged**